src/index.ts

```diff
@@ -13,76 +13,19 @@ You should have received a copy of the GNU Affero General Public License along w
 If not, see <https://www.gnu.org/licenses/>.
 */
 
-import { config } from "dotenv";
-config();
+import "dotenv/config";
+import license from "./helpers/license";
+import ObjectFactory from "./ObjectFactory";
 
-import { ActivityType, Client, GatewayIntentBits } from "discord.js";
-import { getOrLoadAllCommands } from "./handlers/commandHandler";
-import { handleInteractionCreate } from "./handlers/interactionHandler";
-import { handleMessageCreate } from "./handlers/messageHandler";
-import { deleteConfigsFromUnknownServers, getApiToken, resetConfigToDefault } from "./helpers/configHelpers";
-
-console.log(`Needle, a Discord bot that declutters your server by creating threads
-Copyright (C) 2022  Marcus Otterström
-
-This program is free software: you can redistribute it and/or modify
-it under the terms of the GNU Affero General Public License as published
-by the Free Software Foundation, either version 3 of the License, or
-(at your option) any later version.
-
-This program is distributed in the hope that it will be useful,
-but WITHOUT ANY WARRANTY; without even the implied warranty of
-MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-GNU Affero General Public License for more details.
-
-You should have received a copy of the GNU Affero General Public License
-along with this program.  If not, see <https://www.gnu.org/licenses/>.
-`);
+console.log(license);
+const bot = ObjectFactory.createNeedleBot();
 
 (async () => {
```

Comment:
I think we can use top-level await instead because TypeScript considers this a module.

Reply:
Not with our config, not going to investigate further in this PR since changing the module type seems risky atm
```
src/index.ts:23:1 - error TS1378: Top-level 'await' expressions are only allowed when the 'module' option is set to 'es2022', 'esnext', 'system', 'node16', or 'nodenext', and the 'target' option is set to 'es2017' or higher.
```